src/xdocs/config_naming.xml

```diff
@@ -2293,6 +2293,135 @@ class MyClass {
       </subsection>
     </section>
 
+    <section name="PatternVariableName">
+          <p>Since Checkstyle 8.35</p>
+          <subsection name="Description" id="PatternVariableName_Description">
+              <p>
+                  Checks that pattern variable names conform to a specified pattern.
+              </p>
+          </subsection>
+
+          <subsection name="Properties" id="PatternVariableName_Properties">
+              <div class="wrapper">
+                  <table>
+                      <tr>
+                          <th>name</th>
+                          <th>description</th>
+                          <th>type</th>
+                          <th>default value</th>
+                          <th>since</th>
+                      </tr>
+                      <tr>
+                          <td>format</td>
+                          <td>Specifies valid identifiers.</td>
+                          <td><a href="property_types.html#regexp">Pattern</a></td>
+                          <td><code>"^[a-z][a-zA-Z0-9]*$"</code></td>
+                          <td>3.0</td>
+                      </tr>
+                  </table>
+              </div>
+          </subsection>
+
+          <subsection name="Examples" id="PatternVariableName_Examples">
+              <p>
+                  An example of how to configure the check is:
+              </p>
+
+              <source>
+                  &lt;module name="PatternVariableName"/&gt;
+              </source>
+              <p>Code Example:</p>
+              <source>
+                  class MyClass {
+                  {
+                  Object o1 = null;
+                  if (o1 instanceof String STRING) { // violation, name 'STRING' must
+                  // match pattern '^[a-z][a-zA-Z0-9]*$'
+                  }
+                  if (o1 instanceof Integer num) { // OK
+                  }
+                  }
+                  }
```

Comment:
These examples look a little strange - indentation, and curious if there's a reason the code here (and in the xpath example files) is in a block like this and not a constructor or method?

Reply:
I might vote to go ahead and name the block as constructor or method just because it's a little more familiar, but it's a minor point of course -- I'll send this PR to @pbludov now, maybe he has a thought on it.

Reply:
I agree with @esilkensen. The examples should be easy to read. The indentations are definitely necessary.

Reply:
I have pushed since this review, and fixed the indentation mistake. Just to be clear, you would like me to add  methods or constructors instead of just having the code in a static block?

Reply:
In my opinion, the constructor is better .

Reply:
Fixed.